File: src/qmd-rust/src/cli/status.rs
use crate::anel::AnelSpec;
use crate::cli::StatusArgs;
use crate::store::Store;
use anyhow::Result;

/// Handle status command - show index status
pub fn handle(
    cmd: &StatusArgs,
    store: &Store,
) -> Result<()> {
    // Handle --emit-spec: output ANEL specification and exit
    if cmd.emit_spec {
        let spec = AnelSpec::status();
        println!("{}", serde_json::to_string_pretty(&spec)?);
        return Ok(());
    }

    // Handle --dry-run: validate parameters without executing
    if cmd.dry_run {
        println!("[DRY-RUN] Would execute status with:");
        println!("  verbose: {}", cmd.verbose);
        println!("  collection: {:?}", cmd.collection);
        return Ok(());
    }

    println!("QMD Index Status");
    println!("{}", "=".repeat(50));

    let stats = store.get_stats()?;

    println!("\nCollections: {}", stats.collection_count);
    println!("Documents: {}", stats.document_count);
    println!("Indexed: {}", stats.indexed_count);
    println!("Pending: {}", stats.pending_count);

    if cmd.verbose {
        println!("\nDetailed Statistics:");
        for (name, count) in &stats.collection_stats {
            println!("  {}: {} documents", name, count);
        }
    }

    Ok(())
}

File: qmd-go/pkg/cli/status.go
package cli

import (
	"encoding/json"
	"fmt"
	"os"

	"github.com/spf13/cobra"
)

var statusCmd = &cobra.Command{
	Use:   "status",
	Short: "Show index status",
	Run: func(cmd *cobra.Command, args []string) {
		s, err := getStore()
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
			os.Exit(1)
		}
		defer s.Close()

		stats, err := s.GetStats()
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
			os.Exit(1)
		}

		if outputFormat == "json" {
			json.NewEncoder(os.Stdout).Encode(stats)
		} else {
			fmt.Printf("Collections: %d\n", stats.CollectionCount)
			fmt.Printf("Documents: %d\n", stats.DocumentCount)
			fmt.Printf("Indexed: %d\n", stats.IndexedCount)
			fmt.Printf("Chunks: %d\n", stats.ChunkCount)
			fmt.Println("\nPer-collection:")
			for name, count := range stats.CollectionStats {
				fmt.Printf("  %s: %d\n", name, count)
			}
		}
	},
}
